plugins/modules/mysql_info.py

```diff
@@ -738,10 +755,16 @@ def main():
                'Exception message: %s' % (connector_name, connector_version, config_file, to_native(e)))
         module.fail_json(msg)
 
+    cursor.execute("SELECT VERSION()")
+    if 'mariadb' in cursor.fetchone()["VERSION()"].lower():
+        server_implementation = "mariadb"
+    else:
+        server_implementation = "mysql"
+
```

Comment:
This could use `get_server_version()` from module_utils/mysql.py.

Reply:
Just moved it to the `module_utils/mysql.py` and using the `get_server_version()` instead of running the `SELECT VERSION()`